mem/metadata.go

```diff
@@ -0,0 +1,45 @@
+package mem
```

Comment:
maybe this could go in the `peerstore` package as the default implementation for now

Reply:
Is there a particular benefit we can think of putting the mem impl in the outer package? Right now it's nicely namespaced, so a consumer explicitly imports `mem` and constructs a `mem.NewPeerstore()`.